meow/structures.py

```diff
@@ -41,13 +90,44 @@ def visualize_structures(structures, scale=None):
     return scene.show()
 
 
-def sort_structures(structures):
+@overload
+def sort_structures(structures: List[Structure3D]) -> List[Structure3D]:
+    ...
+
+
+@overload
+def sort_structures(structures: List[Structure2D]) -> List[Structure2D]:
+    ...
+
+
+def sort_structures(
+    structures: Union[List[Structure3D], List[Structure2D]]
+) -> Union[List[Structure2D], List[Structure3D]]:
     struct_info = [(s.mesh_order, -i, s) for i, s in enumerate(structures)]
     sorted_struct_info = sorted(struct_info, key=lambda I: (I[0], I[1]), reverse=True)
-    return [s for _, _, s in sorted_struct_info]
+    return [s for _, _, s in sorted_struct_info]  # type: ignore
+
+
+@overload
+def classify_structures_by_mesh_order_and_material(
+    structures: List[Structure3D], materials: Dict[Material, int]
+) -> Dict[Tuple[int, int], List[Structure3D]]:
+    ...
+
+
+@overload
+def classify_structures_by_mesh_order_and_material(
+    structures: List[Structure2D], materials: Dict[Material, int]
+) -> Dict[Tuple[int, int], List[Structure2D]]:
+    ...
 
 
-def classify_structures_by_mesh_order_and_material(structures, materials):
+def classify_structures_by_mesh_order_and_material(
+    structures: Union[List[Structure3D], List[Structure2D]],
+    materials: Dict[Material, int],
+) -> Union[
+    Dict[Tuple[int, int], List[Structure2D]], Dict[Tuple[int, int], List[Structure3D]]
+]:
     structures = sort_structures(structures)
     structures_dict = {}
```

Comment:
The order of this dict determines the layering of structures while building the material matrix. It should this be stored in a datastructure, that respects order (e.g. `OrderedDict`)

Reply:
a python dictionary is guarenteed to be ordered since python 3.6

Reply:
Same here